src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -7295,6 +7294,40 @@ private void SaveError(IScriptExtent extent, string errorId, string errorMsg, bo
             SaveError(errorToSave);
         }
 
+        /// <summary>
+        /// Assertion to ensure that .resx resource files are where all errors
+        /// saved by the parser are stored. Goes through the known resource types
+        /// and checks that the given error ID and message match some entry in one of them
+        /// </summary>
+        /// <param name="errorId">The error ID string (.resx key)</param>
+        /// <param name="errorMsg">The error message, which may be a template string (.resx value)</param>
+        [System.Diagnostics.Conditional("DEBUG")]
+        [System.Diagnostics.Conditional("ASSERTIONS_TRACE")]
+        private static void AssertErrorIdCorrespondsToMsgString(string errorId, string errorMsg)
+        {
+            Type[] resxTypes = new []
+            {
+                typeof(ParserStrings),
+                typeof(DiscoveryExceptions),
+                typeof(ExtendedTypeSystem),
+                typeof(MshSnapInCmdletResources),
+                typeof(ParameterBinderStrings)
+            };
+
+            bool msgCorrespondsToString = false;
+            foreach (Type resxType in resxTypes)
+            {
+                string resxErrorBody = resxType.GetProperty(errorId, BindingFlags.Static | BindingFlags.NonPublic)?.GetValue(null) as string;
+                if (String.Equals(errorMsg, resxErrorBody))
```

Comment:
Maybe use `Ordinal` comparision?

Reply:
Oh -- quite right! We want `StringComparison.CurrentCulture` I think, since they're resource files. Does that sound right?

Reply:
`String.Equals` would be correct (you want an exact comparison - the string should come from the resource, not user generated) but FxCop does complain when you omit the `StringComparison`, so you should use `StringComparison.Ordinal`.